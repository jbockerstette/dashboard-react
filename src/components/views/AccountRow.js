import React from 'react';
import PropTypes from 'prop-types';
import Text from './Text';
import IconButton from './IconButton';
import './AccountRow.css';

const AccountRow = props => {
  const { account, isAvailable } = props;
  const color = isAvailable ? 'primary' : 'disabled';
  const status = isAvailable ? 'Available' : 'Not Available';
  const btnText = isAvailable ? 'Edit' : 'Delete';
  return (
    <div className="AccountRow">
      <div>
        <Text color={color}>{account}</Text>
        <Text htmlTag="div" small color={color}>
          {status}
        </Text>
      </div>
      <IconButton color="red">{btnText}</IconButton>
    </div>
  );
};

AccountRow.propTypes = {
  account: PropTypes.string.isRequired,
  isAvailable: PropTypes.bool
};

AccountRow.defaultProps = {
  isAvailable: false
};

export default AccountRow;
