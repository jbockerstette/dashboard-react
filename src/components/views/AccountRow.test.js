import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import AccountRow from './AccountRow';

const account = '1-123-123-1234';

it('renders without crashing', () => {
  shallow(<AccountRow account={account} />);
});

test('AccountRow renders correctly when not available', () => {
  const tree = renderer.create(<AccountRow account={account} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('AccountRow renders correctly when is available', () => {
  const tree = renderer
    .create(<AccountRow account={account} isAvailable />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
