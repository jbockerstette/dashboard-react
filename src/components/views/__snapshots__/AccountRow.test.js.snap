// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountRow renders correctly when is available 1`] = `
<div
  className="AccountRow"
>
  <div>
    <span
      className="Text-title Text--primary "
    >
      1-123-123-1234
    </span>
    <div
      className="Text-title Text--primary Text--small"
    >
      Available
    </div>
  </div>
  <div>
    <button
      className="Btn Btn--red"
      onClick={null}
    >
      Edit
    </button>
  </div>
</div>
`;

exports[`AccountRow renders correctly when not available 1`] = `
<div
  className="AccountRow"
>
  <div>
    <span
      className="Text-title Text--disabled "
    >
      1-123-123-1234
    </span>
    <div
      className="Text-title Text--disabled Text--small"
    >
      Not Available
    </div>
  </div>
  <div>
    <button
      className="Btn Btn--red"
      onClick={null}
    >
      Delete
    </button>
  </div>
</div>
`;
